specification/dns/resource-manager/Microsoft.Network/stable/2017-10-01/dns.json

```diff
@@ -0,0 +1,1584 @@
+          },
+          {
+            "name": "zoneName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the DNS zone (without a terminating dot)."
+          },
+          {
+            "name": "$top",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "description": "The maximum number of record sets to return. If not specified, returns up to 100 record sets."
+          },
+          {
+            "name": "$recordsetnamesuffix",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The suffix label of the record set name that has to be used to filter the record set enumerations. If this parameter is specified, Enumeration will return only records that end with .<recordSetNameSuffix>"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/RecordSetListResult"
+            }
+          },
+          "default": {
+            "description": "Default response. It will be deserialized as per the Error definition.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List recordsets by zone": {
+            "$ref": "./examples/ListRecordSetsByZone.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnsZones/{zoneName}/all": {
+      "get": {
+        "tags": [
+          "RecordSets"
+        ],
+        "operationId": "RecordSets_ListAllByDnsZone",
+        "description": "Lists all record sets in a DNS zone.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "zoneName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the DNS zone (without a terminating dot)."
+          },
+          {
+            "name": "$top",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "description": "The maximum number of record sets to return. If not specified, returns up to 100 record sets."
+          },
+          {
+            "name": "$recordsetnamesuffix",
```

Comment:
>$recordsetnamesuffix [](start = 21, length = 20)

For this param in particular it would be a good idea to use x-ms-client-name extension

Reply:
Yes, this is a good suggestion. I am going to fix it for this new API "/all". It looks like somehow 2 existing APIs slipped through the cracks with this same parameter without this unfortunately. I am going to avoid making a breaking SDK change on the existing 2 for now, unless you feel there is a compelling case here for a breaking SDK change.

Reply:
Oh no, lets not create breaking changes just for changing a name :)